### DISCUSSION TITLE
What type should I apply to the options parameter?

### DISCUSSION THREAD

**Discussion by JustDenP** [mikro-orm/mikro-orm#3678] [Q&A] [answered]
I have a find functions that I call from different places. Depending on the situation, I need to add different options, for example populate, but no matter how I try and what types I apply, I always get the error "Type 'string' is not assignable to type 'never'".

It works fine without the type, but I get no help and no checks from typescript when passing the information into options.

Could you please tell me how to correctly assign a type to it? Thank you!
This is examples of my functions:

Function inside user.service.ts
```typescript
async findByEmail(
    email: string,
    options?: FindOneOptions<UserEntity>
  ): Promise<UserEntity | null> {
    return this.userRepository.findOne(
      {
        email,
      },
      options,
    );
  }
```

Then inside my controller file I want to use this function and plus load the related entitiy "settings"
```typescript
const user = await this.userService.findByEmail(userData.email, {
   populate: ['settings'],
});
```
This populate field always gives me an error "Type 'string' is not assignable to type 'never'"

**Comment by B4nan** [comment, marked as answer]
Same way as the method you are wrapping, in this case `repo.findOne`:

```ts
async findByEmail<P extends string = never>(email: string, options?: FindOneOptions<UserEntity, P>): Promise<Loaded<UserEntity, P> | null> {
  return this.userRepository.findOne<P>({ email }, options);
}
```

**Reply by JustDenP** [reply]
Thank you very much, I'm not very good at typescript :)
That solved the problem.
